mmv1/third_party/terraform/utils/provider.go.erb

```diff
@@ -496,6 +496,12 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
         <% end -%>
 	}
 
+	// opt in extension for adding to the User-Agent header
+	if ext := os.Getenv("GOOGLE_TPG_USERAGENT_EXTENSION"); ext != "" {
```

Comment:
@rileykarson is the env variable name okay?

`GOOGLE_TERRAFORM_PROVIDER_USERAGENT_EXTENSION` was too wordy
`TPG_USERAGENT_EXTENSION` was to ambiguous

Reply:
wdyt of just `GOOGLE_TERRAFORM_USERAGENT_EXTENSION`? Since this'll apply across multiple providers.

Reply:
i dig it